test/test_core.py

```diff
@@ -2884,7 +2884,7 @@ def test_pthread_abort(self):
 
   @node_pthreads
   def test_pthread_abort_interrupt(self):
-    self.set_setting('PTHREAD_POOL_SIZE', 1)
+    self.set_setting('PROXY_TO_PTHREAD')
```

Comment:
It's indeed a bit curious that this is needed. Linking with `-sPTHREAD_POOL_SIZE=1` and `-sEXIT_RUNTIME` doesn't work either.

Reply:
This particular test case is calling `abort()` in its pthread entry point, which causes a `WebAssembly.RuntimeError` to be thrown that is supposed to be propagated back to the main thread via the [`'error'` event](https://nodejs.org/api/worker_threads.html#event-error). However, the worker is calling `_emscripten_thread_crashed()` just before throwing this exception:
https://github.com/emscripten-core/emscripten/blob/574f4f449780f63ab97970d5d6b5625e124c7d79/src/worker.js#L278-L281

As a result, the main thread immediately exits while the runtime is kept alive:
https://github.com/emscripten-core/emscripten/blob/574f4f449780f63ab97970d5d6b5625e124c7d79/system/lib/pthread/emscripten_yield.c#L31-L35

Since the worker is weakly referenced, the program exits too early, causing the test to fail.

---

A possible solution to this is to make the thread strongly referenced during `pthread_join()`:
https://github.com/emscripten-core/emscripten/blob/42df09ece7401602a085992784346c166cc8e4a3/system/lib/libc/musl/src/thread/pthread_join.c#L21-L23

This ensures that the program does not exit too early when a joinable thread crashes.

Reply:
Commit ca5f97c364baa60b7a4699f47916a2e3ed2afe5e improves this.